objectstore-service/src/lib.rs

```diff
@@ -73,16 +74,33 @@ impl StorageService {
         usecase: String,
         scope: Scope,
         metadata: &Metadata,
-        stream: BackendStream,
+        mut stream: BackendStream,
     ) -> anyhow::Result<ScopedKey> {
-        let key = ObjectKey::for_backend(1);
+        let mut first_chunk = BytesMut::new();
+        let mut backend_id = 1; // 1 = small files backend
+        while let Some(chunk) = stream.try_next().await? {
+            first_chunk.extend_from_slice(&chunk);
+
+            if first_chunk.len() > SMALL_THRESHOLD {
+                backend_id = 2; // 2 = large files backend
+                break;
+            }
```

Comment:
We need to ensure in the interface that overwrites of _existing_ objects in the same backend are possible. I will finish the [decision on object mutability](https://www.notion.so/sentry/Are-objects-mutable-25d8b10e4b5d8019b92cf30c9acc5b47?source=copy_link) today - expect that we will go with option 3 at the bottom:

> Allow payload updates with the same ID and actually replace the data (or use versioning in the backend where available). If the new payload exceeds hard limits of the backend that the object was previously stored in, we reject the update.

This implies that we also need to add a second hard limit for when to reject the update. Example with real values:
- Threshold for BigTable -> GCS is 1MB. 
- Hard limit for BigTable is 10MB.